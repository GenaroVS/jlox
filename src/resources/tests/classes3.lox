// init method edge cases

class Point {

    init(x, y) {
        this.x = x;
        this.y = y;
        return ;
    }

    add(point) {
        return Point(this.x + point.x, this.y + point.y);
    }
}

var p1 = Point(2, 3);

print "Initializing";
print "X";
print p1.x;
print "Y";
print p1.y;

p1 = p1.init(4, 5);

print "Re-initializing";
print "X";
print p1.x;
print "Y";
print p1.y;