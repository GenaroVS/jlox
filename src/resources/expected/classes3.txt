Initializing
X
2
Y
3
Re-initializing
X
4
Y
5
